mmv1/third_party/terraform/resources/resource_bigtable_instance.go

```diff
@@ -260,7 +262,17 @@ func resourceBigtableInstanceRead(d *schema.ResourceData, meta interface{}) erro
 
 	clusters, err := c.Clusters(ctx, instance.Name)
 	if err != nil {
-		return fmt.Errorf("Error retrieving instance clusters. %s", err)
+		partiallyUnavailableErr, ok := err.(bigtable.ErrPartiallyUnavailable)
+
+		if !ok {
+			return fmt.Errorf("Error retrieving instance clusters. %s", err)
+		}
+
+		unavailableClusterZones := getUnavailableClusterZones(d.Get("cluster").([]interface{}), partiallyUnavailableErr.Locations)
```

Comment:
The proposed PR seems to accept the possibility that some cluster was created in unavailable region outside of Terraform and will only now show up when calling c.Clusters() for a given instance. Do you think it's safe?

Could it potentially lead to some cluster deleted if we proceed with this reduced state of clusters and pass it to our Go client? 

My main concern is that if go client doesn't see a cluster in Paris in the desired configuration, an update with this reduced target configuration might later lead to cluster in Paris deleted?

Reply:
Any potential change actually goes the _other_ way. Because the call to c.Clusters returned an error, Terraform will only check that clusters currently recorded in state are unavailable. If a cluster was added out of band, Terraform will proceed with the read. That's probably a rare case, though, since most users will manage their whole state with Terraform.

Returning an error here is fine- that's the same as saying an API returned a 5XX error in HTTP, Terraform just understands itself to have failed to read the resource. If we unset the id it could cause problem (that means "failed to read because the resource no longer exists")

Reply:
If a cluster was added out of band in a zone that is now unavailable, it would be the same as if they added it out of band in a zone that _was_ available - Terraform might want to remove it on the following apply. But that wouldn't be a result of this PR.